lib/Sema/SemaBounds.cpp

```diff
@@ -932,6 +943,19 @@ namespace {
         return true;
       }
 
+      if (CK == CK_PointerBounds &&
+          (dyn_cast<BoundsCastExpr>(E)->getBoundsCastKind() ==
+           BoundsCastExpr::Kind::Dynamic)) {
+        BoundsExpr *SrcBounds = S.InferRValueBounds(E);
+        if (SrcBounds->isNone()) {
```

Comment:
I'm really sorry that this is mixed with previous code. I changed CK_PointerBounds to CK_DynamicPtrBounds and CK_AssumePtrBounds. I think it is not updated. I will Again.